torch/distributed/tensor/parallel/fsdp.py

```diff
@@ -184,7 +184,9 @@ def _create_sharded_tensor_md_from_dt(
 def _get_dt_pg(dt: DistributedTensor) -> c10d.ProcessGroup:
     mesh = dt.device_mesh
     assert mesh.ndim == 1, "Only 1D DeviceMeshes currently handled"
-    return mesh.get_dim_groups()[0]
+    dim_groups = mesh.get_dim_groups()
+    assert isinstance(dim_groups, List[c10d.ProcessGroup])
```

Comment:
`assert isinstance(..., List)` but we better get rid of checking types.

Reply:
there's no way to get rid of them if making `get_dim_groups` return two different types, and I think type check is a good thing for python program to make sure there's no apparent bugs